include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,918 @@
+    using size_type       = detail::min_viable_uint_t<N>;     //!< The size_type type.
+    //!\}
+
+    //!\cond
+    // this signals to range-v3 that something is a container :|
+    using allocator_type    = void;
+    //!\endcond
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr constexpr_vector() = default;                                     //!< Defaulted.
+    constexpr constexpr_vector(constexpr_vector const &) = default;             //!< Defaulted.
+    constexpr constexpr_vector(constexpr_vector &&) = default;                  //!< Defaulted.
+    constexpr constexpr_vector & operator=(constexpr_vector const &) = default; //!< Defaulted.
+    constexpr constexpr_vector & operator=(constexpr_vector &&) = default;      //!< Defaulted.
+    ~constexpr_vector() = default;                                              //!< Defaulted.
+
+    /*!\brief Construct from an (smaller or equally sized) array over the same value type.
+     * \param[in] array The array to copy the data from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `array`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr constexpr_vector(std::array<value_type, N> const & array) :
+        data{array}, sz{N}
+    {}
+
+    //!\cond
+    template <size_t N_other>
+    //!\cond
+        requires (N_other <= N)
+    //!\endcond
+    constexpr constexpr_vector(std::array<value_type, N_other> const & array) :
+        sz{N_other}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+    //!\endcond
+
+    /*!\brief Construct from a (smaller or equally sized) built in array over the same value type.
+     * \param[in] array The array to copy the data from.
+     *
+     * ### Complexity
+     *
+     * Linear in `N`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr constexpr_vector(const value_type (&array)[N]) :
+        sz{N}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+
+    //!\cond
+    template <size_t N_other>
+    //!\cond
+        requires (N_other <= N)
+    //!\endcond
+    constexpr constexpr_vector(const value_type (&array)[N_other]) :
+        sz{N_other}
+    {
+        std::ranges::copy(array, data.begin());
+    }
+    //!\endcond
+
+    /*!\brief Construct from a list of values of value_type.
+     * \tparam other_value_type A parameter pack where each type is equal to value_type.
+     * \param[in] args The values to construct from.
+     */
+    template <typename ...other_value_type>
+    //!\cond
+        requires (std::Same<value_type, other_value_type> && ...)
+    //!\endcond
+    constexpr constexpr_vector(other_value_type... args) :
+        data{args...}, sz{sizeof...(other_value_type)}
+    {}
```

Comment:
Question: is the static_assert within the constructor body evaluated before member initialization happens? Because then `data{args...}` might fail before the static_assert catches.

Reply:
You did not answer by question :(

Reply:
> Question: is the static_assert within the constructor body evaluated before member initialization happens?

I don't know.

> Because then data{args...} might fail before the static_assert catches.

Yes, that's possible.

Reply:
I tried it. The static assert catches before the member initialization is done